src/policy.ts

```diff
@@ -0,0 +1,178 @@
+import {
+  Braid,
+  validateBIP32Path,
+  validateExtendedPublicKey,
+  validateRootFingerprint,
+} from "unchained-bitcoin";
+import { WalletPolicy } from "./vendor/ledger-bitcoin";
+import { BitcoinNetwork } from "./types";
+
+export class KeyOrigin {
+  xfp: string;
+
+  bip32Path: string;
+
+  xpub: string;
+
+  network: BitcoinNetwork;
```

Comment:
Same as in `ledger.ts`, maybe this needs to use more typescript class features. They would be hard to add later if these get used in unintended ways.